weather@mockturtl/files/weather@mockturtl/3.4/applet.js

```diff
@@ -0,0 +1,2345 @@
+"use strict";
+
+// Import Polyfills
+/*var AppletDir = imports.ui.appletManager.applets['weather@mockturtl'];
+var regenerator = AppletDir.regenerator;
+var regeneratorRuntime = regenerator.runtime;*/
+
+// Polyfill for Array.includes for 3.6
+if (!Array.prototype.includes) {
+  Object.defineProperty(Array.prototype, 'includes', {
+    value: function(valueToFind, fromIndex) {
+
+      if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+      }
+
+      // 1. Let O be ? ToObject(this value).
+      var o = Object(this);
+
+      // 2. Let len be ? ToLength(? Get(O, "length")).
+      var len = o.length >>> 0;
+
+      // 3. If len is 0, return false.
+      if (len === 0) {
+        return false;
+      }
+
+      // 4. Let n be ? ToInteger(fromIndex).
+      //    (If fromIndex is undefined, this step produces the value 0.)
+      var n = fromIndex | 0;
+
+      // 5. If n ≥ 0, then
+      //  a. Let k be n.
+      // 6. Else n < 0,
+      //  a. Let k be len + n.
+      //  b. If k < 0, let k be 0.
+      var k = Math.max(n >= 0 ? n : len - Math.abs(n), 0);
+
+      function sameValueZero(x, y) {
+        return x === y || (typeof x === 'number' && typeof y === 'number' && isNaN(x) && isNaN(y));
+      }
+
+      // 7. Repeat, while k < len
+      while (k < len) {
+        // a. Let elementK be the result of ? Get(O, ! ToString(k)).
+        // b. If SameValueZero(valueToFind, elementK) is true, return true.
+        if (sameValueZero(o[k], valueToFind)) {
+          return true;
+        }
+        // c. Increase k by 1. 
+        k++;
+      }
+
+      // 8. Return false
+      return false;
+    }
+  });
+}
+
+// Polyfill for Babel async: regenerator-runtime
+/**
+ * Copyright (c) 2014-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+var runtime=function(g){function D(a,b,c,e){b=Object.create((b&&b.prototype instanceof y?b:y).prototype);e=new z(e||[]);b._invoke=K(a,c,e);return b}function A(a,b,c){try{return{type:"normal",arg:a.call(b,c)}}catch(e){return{type:"throw",arg:e}}}function y(){}function t(){}function q(){}function E(a){["next","throw","return"].forEach(function(b){a[b]=function(a){return this._invoke(b,a)}})}function u(a){function b(c,d,l,f){c=A(a[c],a,d);if("throw"===c.type)f(c.arg);else{var e=c.arg;return(c=e.value)&&
```

Comment:
I would wait until Cinnamon is using mozjs60 or newer before using async/await. regenerator-runtime will add quite a bit of overhead, and we can't forget the process is also managing compositing and file I/O.

Reply:
I will not use babel and regenerator but I could transpile the code from typescript. It uses its own generator (smaller).

This is obviously not ideal, but people's main concern that OpenWeatherMap is not accurate enough, and there is no other option below 3.8. Hence the pr as I know I don't have time/won't be able to do the same with Promises reliably.